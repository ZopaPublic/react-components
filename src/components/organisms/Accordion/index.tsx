import AccordionComponent from './Accordion/Accordion';
import AccordionHeader from './AccordionHeader/AccordionHeader';
import AccordionSection from './AccordionSection/AccordionSection';

export const Accordion = Object.assign(AccordionComponent, {
  Header: AccordionHeader,
  Section: AccordionSection,
});
