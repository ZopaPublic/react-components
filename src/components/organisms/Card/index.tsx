import React, { HTMLAttributes } from 'react';

import CardActions from './CardActions/CardActions';
import CardComponent, { CardProps } from './Card/Card';
import CardContent from './CardContent/CardContent';
import CardHeading from './CardHeading/CardHeading';
import CardImage from './CardImage/CardImage';
import CardText from './CardText/CardText';
import CardLineItem from './CardLineItem/CardLineItem';

const Card = (props: CardProps & HTMLAttributes<HTMLDivElement>) => <CardComponent {...props} />;

export default Object.assign(Card, {
  Actions: CardActions,
  Content: CardContent,
  Heading: CardHeading,
  Image: CardImage,
  Text: CardText,
  LineItem: CardLineItem,
});
