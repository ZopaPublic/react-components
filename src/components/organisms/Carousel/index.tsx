import CarouselComponent from './Carousel/Carousel';
import Slide from './CarouselSlide/CarouselSlide';
import SlideIcon from './CarouselSlideIcon/CarouselSlideIcon';
import SlideText from './CarouselSlideText/CarouselSlideText';

export const Carousel = Object.assign(CarouselComponent, {
  Slide: Slide,
  SlideIcon: SlideIcon,
  SlideText: SlideText,
});
