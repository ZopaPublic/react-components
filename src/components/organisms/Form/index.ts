export { default as FormButton } from './FormButton/FormButton';
export { default as FormTextField } from './FormTextField/FormTextField';
export { default as FormCheckboxField } from './FormCheckboxField/FormCheckboxField';
export { default as FormCheckboxGroupField } from './FormCheckboxGroupField/FormCheckboxGroupField';
export { default as FormRadioGroupField } from './FormRadioGroupField/FormRadioGroupField';
export { default as FormDropdownField } from './FormDropdownField/FormDropdownField';
export { default as FormDropdownFilteredField } from './FormDropdownFilteredField/FormDropdownFilteredField';
export { default as FormSection } from './FormSection/FormSection';
