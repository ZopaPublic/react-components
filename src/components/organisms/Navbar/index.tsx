import NavbarWrapper from './Navbar/Navbar';
import NavbarAction from './NavbarAction/NavbarAction';

export default Object.assign(NavbarWrapper, { Action: NavbarAction });

export { navbarLinkStyles } from './NavbarLink/NavbarLink';
