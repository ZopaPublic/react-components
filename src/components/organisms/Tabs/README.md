### Summary

`<Tabs />` is a wrapper that is supposed to be used with `<Tabs.Buttons />` and `<Tabs.Content />`. It provides its children with the `<TabsContext />`.

### Example

```tsx
import { Tabs } from '@zopauk/react-components';

function TabsExample() {
  const tabButtons = [
    { tabId: 'pineapple', title: '🍍 Pineapple' },
    { tabId: 'kiwi', title: '🥝 Kiwi' },
    {
      tabId: 'watermelon',
      title: '🍉 Watermelon',
      afterOnClick: () => console.log('This tab has a function that is called after the default action in onClick'),
    },
  ];
  return (
    <Tabs>
      <Tabs.Buttons tabButtons={tabButtons} defaultTab="pineapple" />
      <Tabs.Content contentFor="pineapple">🍍</Tabs.Content>
      <Tabs.Content contentFor="kiwi">🥝</Tabs.Content>
      <Tabs.Content contentFor="watermelon">🍉</Tabs.Content>
    </Tabs>
  );
}

<TabsExample />;
```
