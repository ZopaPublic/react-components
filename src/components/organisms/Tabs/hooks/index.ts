export * from './useTabs';
export { TabsContext } from './TabsContext';
export { useTabsContext } from './useTabsContext';
