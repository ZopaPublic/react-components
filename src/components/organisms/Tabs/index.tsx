import React from 'react';

import TabsContainer, { TabsProps } from './TabsContainer/TabsContainer';
import TabContent from './TabContent/TabContent';
import TabButtons from './TabButtons/TabButtons';

const TabsMain = (props: TabsProps) => <TabsContainer {...props} />;

export const Tabs = Object.assign(TabsMain, { Content: TabContent, Buttons: TabButtons });

export * from './hooks';
