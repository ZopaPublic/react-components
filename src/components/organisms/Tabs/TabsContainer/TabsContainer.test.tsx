import React from 'react';
import axe from '../../../../../axe-helper';
import { render } from '@testing-library/react';
import { Tabs } from '../../../organisms/Tabs';

describe('<TabContainer />', () => {
  const tabButtons = [
    { tabId: 'pineapple', title: '🍍 Pineapple' },
    { tabId: 'kiwi', title: '🥝 Kiwi' },
    { tabId: 'watermelon', title: '🍉 Watermelon' },
  ];

  it('renders the component with no a11y violations', async () => {
    const { container } = render(
      <Tabs>
        <Tabs.Buttons tabButtons={tabButtons} defaultTab="pineapple" />
        <Tabs.Content contentFor="pineapple">🍍</Tabs.Content>
        <Tabs.Content contentFor="kiwi">🥝</Tabs.Content>
        <Tabs.Content contentFor="watermelon">🍉</Tabs.Content>
      </Tabs>,
    );
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
