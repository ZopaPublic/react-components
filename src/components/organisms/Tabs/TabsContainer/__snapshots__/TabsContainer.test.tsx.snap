// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabContainer /> renders the component with no a11y violations 1`] = `
.c5 {
  display: none;
}

.c5[id="pineapple-content"] {
  display: block;
}

.c1 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c2 {
  outline: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px 24px;
  background: transparent;
  border-radius: 8px;
  border: 1px solid transparent;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  color: #21247F;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c2:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c2:active {
  opacity: 0.8;
  box-shadow: unset;
  border: 1px solid transparent;
}

.c2:hover {
  background: #EAEBFA;
}

.c3 {
  box-shadow: inset 0 12px #F7F7F7,inset 0 -12px #F7F7F7,inset 1px 0 #D4D7D9;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100%;
  background: #FFFFFF;
  box-shadow: none;
}

.c3:first-child {
  box-shadow: none;
}

.c3:hover,
.c3:focus {
  box-shadow: none;
}

.c3:hover + button,
.c3:focus + button {
  box-shadow: none;
}

.c3 + button {
  box-shadow: none;
}

.c4 {
  box-shadow: inset 0 12px #F7F7F7,inset 0 -12px #F7F7F7,inset 1px 0 #D4D7D9;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100%;
}

.c4:first-child {
  box-shadow: none;
}

.c4:hover,
.c4:focus {
  box-shadow: none;
}

.c4:hover + button,
.c4:focus + button {
  box-shadow: none;
}

.c0 {
  background: #F7F7F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

@media (min-width:768px) {
  .c1 {
    display: block;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div>
  <div
    class="c0 p-2"
  >
    <div
      class="c1"
    >
      <button
        aria-controls="pineapple-content"
        aria-expanded="true"
        class="c2 c3"
        id="pineapple"
        styling="link"
        theme="[object Object]"
      >
        üçç Pineapple
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        aria-controls="kiwi-content"
        aria-expanded="false"
        class="c2 c4"
        id="kiwi"
        styling="link"
        theme="[object Object]"
      >
        ü•ù Kiwi
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        aria-controls="watermelon-content"
        aria-expanded="false"
        class="c2 c4"
        id="watermelon"
        styling="link"
        theme="[object Object]"
      >
        üçâ Watermelon
      </button>
    </div>
  </div>
  <div
    aria-hidden="false"
    aria-labelledby="pineapple"
    class="c5"
    id="pineapple-content"
    role="region"
  >
    üçç
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="kiwi"
    class="c5"
    id="kiwi-content"
    role="region"
  >
    ü•ù
  </div>
  <div
    aria-hidden="true"
    aria-labelledby="watermelon"
    class="c5"
    id="watermelon-content"
    role="region"
  >
    üçâ
  </div>
</div>
`;
